src/routes/pat/pat.ts

```diff
@@ -0,0 +1,60 @@
+import { sendError } from '@/errors';
+import { getUser, gqlSdk } from '@/utils';
+import { RequestHandler } from 'express';
+import Joi from 'joi';
+import { v4 as uuidv4 } from 'uuid';
+
+export const createPATSchema = Joi.object({
+  expiresAt: Joi.date().required(),
+  metadata: Joi.object(),
+}).meta({ className: 'CreatePATSchema' });
+
+export const createPATHandler: RequestHandler<
+  {},
+  {},
+  { metadata: object; expiresAt: Date }
+> = async (req, res) => {
+  if (!req.auth) {
+    return sendError(res, 'unauthenticated-user');
+  }
+
+  const { userId } = req.auth as RequestAuth;
+
+  const user = await getUser({ userId });
+
+  if (!user) {
+    return sendError(res, 'invalid-refresh-token');
+  }
+
+  const { metadata, expiresAt } = req.body;
+
+  // Note: Joi wouldn't work here because we need to compare the date to the
+  // date of the request, not the date when the schema was created
+  // 7 days
```

Comment:
out of curiosity, what is the reasoning for 7 days?

Reply:
I followed GitHub's approach in this. The minimum amount of expiration for their legacy tokens is 7 days. We can change it to anything.

Reply:
:+1: